specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json

```diff
@@ -625,6 +625,7 @@
       "name": "ruleName",
       "in": "path",
       "required": true,
+      "pattern": "^.*$",
```

Comment:
>^.*$ [](http://example.com/codeflow?start=18&length=4)

Looks like this is matching any string. 

Is this still needed ? 

Also, if there were some weird strings that got accepted as a ruleName and are provisioned, what happens to them now ? are they blocked ?

Same for all places using this pattern.

Reply:
@ramoka178 Fixed to use a regex that is aligned with our resource naming validation